### DISCUSSION TITLE
Build error in bindgen.

### DISCUSSION THREAD

**Discussion by d3zd3z** [zephyrproject-rtos/zephyr-lang-rust#11] [Q&A]
any hints on build? I get below error

```
error: RUST (defined at modules/Kconfig.rust:4) is assigned in a configuration file, but is not
directly user-configurable (has no prompt). It gets its value indirectly from other symbols. See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_RUST and/or look up RUST in the
menuconfig/guiconfig interface. The Application Development Primer, Setting Configuration Values,
and Kconfig - Tips and Best Practices sections of the manual might be helpful too.
```

**Comment by d3zd3z** [comment]
Are you using the tip of `main` in the zephyr branch? It needs to be based on the very latest code there for this to work. The module most be mentioned in one of the manifest files.

**Reply by hakehuang** [reply]
I am using the latest zephyr main branch code.  here is my steps:
```
Install rust environ
https://www.digitalocean.com/community/tutorials/install-rust-on-ubuntu-linux
          rustup target add riscv32i-unknown-none-elf
          rustup target add riscv64imac-unknown-none-elf
          rustup target add thumbv6m-none-eabi
          rustup target add thumbv7em-none-eabi
          rustup target add thumbv7m-none-eabi
          rustup target add thumbv8m.main-none-eabi
 
	Ø west init .
	Ø west update
	Ø git clone https://github.com/zephyrproject-rtos/zephyr-lang-rust.git 

Cd to your zephr repo path
	Ø source ./zephyr.sh
	Ø west twister  -T ../zephyr-lang-rust/samples/ --build-only

```

**Reply by d3zd3z** [reply]
So, with the repo cloned like this, it won't be considered a module. The rust module is optional (unfortunately), so that needs to be enabled for west to pull it in:

```
west config manifest.group-filter -- +optional
west update
```

After this, check if `../modules/lang/rust` got pulled. This will be a git repo, and you can pull recent versions, or PRs to test, etc.

But, then you should be able to do:

```
west twister -T ../modules/lang/rust/samples/ --build-only
```

and the likes.

## Or use own module
You could also make the module where you want it to add to your module path. I'm not sure how to do this, as I've always just added the module to one of the manifests.

**Reply by hakehuang** [reply]
Thanks, it works, but why it does not working if I just 
'''
git clone https://github.com/zephyrproject-rtos/zephyr-lang-rust.git
cd  zephyr-lang-rust
west twister -T samples/ --build-only
'''

**Reply by d3zd3z** [reply]
The Zephyr build system needs to be aware of the module, otherwise it doesn't use the Kconfig and cmake files from the module. The normal mechanism for that is through the `west.yml` and `submanifests/optional.yaml` files. I think there are ways to manually add to the module path, but I've found using these to be the easiest.

**Reply by adoerr** [reply]
> Are you using the tip of `main` in the zephyr branch? It needs to be based on the very latest code there for this to work. The module most be mentioned in one of the manifest files.

Could you please outline, which mainline features the Rust module depends on specifically? Or phrased differently, how much work would it be to backport to 3.7 :thinking:
